pkg/controller/servicemesh/webhooks/server.go

```diff
@@ -78,5 +86,46 @@ func Add(mgr manager.Manager) error {
 		Handler: validation.NewMemberValidator(),
 	})
 
+	params = &serverParams{
+		ctx: ctx,
+		mgr: mgr,
+	}
+
+	// If SME CRD already exists, go ahead and register the SME conversion webhook right away;
+	// Don't rely on it to be registered on SMCP reconcile, as the SMCP resource might be up to
+	// date therefore the reconcile code won't be called.
+	if crdExists(ctx, mgr, webhookca.ServiceMeshExtensionCRDName) {
+		return RegisterControlPlaneWebhooks()
+	}
+
+	return nil
+}
+
+func RegisterControlPlaneWebhooks() error {
+	if params == nil {
+		log.Info("webhooks.Add() was not called or have failed. Ignoring.")
+		return nil
+	}
+
+	if params.cpwhRegistered {
+		return nil
+	}
+
+	if err := registerConversionWebhook(
+		params.ctx,
+		params.mgr,
+		log,
+		common.GetOperatorNamespace(),
+		&smeConverterServicePath,
+		webhookca.ServiceMeshExtensionCRDName); err != nil {
+		return err
+	}
+
+	log.Info("Adding Maistra ServiceMeshExtension conversion handler")
+	wh := &conversion.Webhook{}
+	wh.InjectScheme(params.mgr.GetScheme())
+	params.mgr.GetWebhookServer().Register(smeConverterServicePath, wh)
```

Comment:
Why wouldn't we continue to register to the server piece at startup?

Reply:
+1